crates/extensions/c8y_mapper_ext/src/fragments.rs

```diff
@@ -36,31 +34,9 @@ impl C8yAgentFragment {
     }
 }
 
-pub fn get_tedge_version() -> Result<String, ConversionError> {
-    let process = if cfg!(test) {
-        assert_cmd::Command::cargo_bin("tedge")
-            .unwrap()
-            .arg("--version")
-            .output()
-    } else {
-        Command::new("tedge").arg("--version").output()
-    };
-
-    match process {
-        Ok(process) => {
-            let string = String::from_utf8(process.stdout)?;
-            Ok(string
-                .split_whitespace()
-                .last()
-                .ok_or(ConversionError::FromOptionError)?
-                .trim()
-                .to_string())
-        }
-        Err(err) => {
-            warn!("{}\ntedge version not found.", err);
-            Ok("unknown".to_string())
-        }
-    }
+pub fn get_tedge_version() -> String {
+    // Use package version over tedge cli to remove dependency on the optional tedge cli #1991
+    env!("CARGO_PKG_VERSION").to_string()
```

Comment:
I'm fine with that (indeed far better than calling tedge cli). However, I wonder if this is on purpose that this differs from `tedge-mapper --version` (the latter using `$GIT_SEMVER`)?

Reply:
Yes good point...though from memory, I think `GIT_SEMVER` is changing the value of the `CARGO_PKG_VERSION` variable if it is defined at build time...so I think we're safe.

e.g. in `crates/core/tedge/build.rs`

```rust
if let Ok(val) = std::env::var("GIT_SEMVER") {
    println!("Using version defined by 'GIT_SEMVER={}'", val);
    println!("cargo:rustc-env=CARGO_PKG_VERSION={}", val);
 }
```

Reply:
Ah...right...I guess since the build.rs is not defined for the crate then it won't be overwritten...let me export that more.

Reply:
@didier-wenzek You were right...since the reference to the `CARGO_PKG_VERSION` variable is referenced from inside its own crate, the version is the version in the `Cargo.toml` and not the `GIT_SEMVER` version.

I'll push the integration test which covers this, and work out how best to resolve the issue.